target/linux/ath79/dts/ar9330_ziking_cpe46b.dts

```diff
@@ -0,0 +1,117 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "ar9330.dtsi"
+
+/ {
+        model = "ZiKing CPE46B";
+        compatible = "ziking,cpe46b", "qca,ar9330";
+
+        aliases {
+                serial0 = &uart;
+                label-mac-device = &eth0;
+        };
+
+        leds {
+                compatible = "gpio-leds";
+
+                wlan {
+                        label = "ziking-cpe46b:green:wlan";
+                        gpios = <&gpio 0 GPIO_ACTIVE_HIGH>;
+                        linux,default-trigger = "phy0tpt";
+                };
+
+                rssi1 {
+                        label = "ziking-cpe46b:green:rssi1";
+                        gpios = <&gpio 20 GPIO_ACTIVE_HIGH>;
+                };
+
+                rssi2 {
+                        label = "ziking-cpe46b:green:rssi2";
+                        gpios = <&gpio 23 GPIO_ACTIVE_HIGH>;
+                };
+        };
+
+        keys {
+                compatible = "gpio-keys";
+
+                reset {
+                        label = "reset";
+                        linux,code = <KEY_RESTART>;
+                        gpios = <&gpio 12 GPIO_ACTIVE_LOW>;
+                };
+        };
+};
+
+&gpio {
+        status = "okay";
+};
+
+&uart {
+        status = "okay";
+};
+
+
+&spi {
+        status = "okay";
+
+        num-chipselects = <1>;
+
+        flash@0 {
+                compatible = "jedec,spi-nor";
+                spi-max-frequency = <50000000>;
+                reg = <0>;
+
+                partitions {
+                        compatible = "fixed-partitions";
+                        #address-cells = <1>;
+                        #size-cells = <1>;
+
+                        partition@0 {
+                                label = "u-boot";
+                                reg = <0x000000 0x010000>;
+                                read-only;
+                        };
+
+                        partition@1 {
+                                label = "u-boot-env";
+                                reg = <0x010000 0x010000>;
+                        };
+
+                        partition@2 {
+                                compatible = "denx,uimage";
+                                label = "firmware";
+                                reg = <0x020000 0x7d0000>;
+                        };
+
+                        art: partition@3 {
+                                label = "art";
+                                reg = <0x7f0000 0x010000>;
+                                read-only;
+                        };
+                };
+        };
+};
+
+&eth0 {
+        status = "okay";
+
+        mtd-mac-address = <&art 0x0>;
+};
+
+&eth1 {
+        status = "okay";
+
+        mtd-mac-address = <&art 0x6>;
```

Comment:
does this correspond to vendor MAC assignment?

Reply:
I have actually an issue there, the vendor shipped ART partition has a placeholder mac address there ("0a  0b 0c 0d 0e 01"). If I run "progmac" from u-boot then it gets rewritten with a valid one (although not the one in the label). I think original vendor firmware just did wlan_mac+1 (since eth0 and wlan mac are as an example :00 and :01 I expect eth2 would be :02). Can you point me in the right direction to fix this?